criu/config.c

```diff
@@ -46,53 +46,57 @@ int parse_statement(int i, char *line, char **configuration)
 {
 	int offset = 0, len = 0;
 	bool was_newline = true;
-	char *tmp_string, *quoted, *quotedptr;
+	char *tmp_string;
 
 	while (1) {
 		/* Ignore white-space */
 		while ((isspace(*(line + offset)) && (*(line + offset) != '\n'))) offset++;
 
+		/* Ignore line starting with a comment */
+		if (line[offset] == '#')
+			break;
+
 		/* Read a single word. A word is everything
 		 * that doesn't contain white-space characters. */
 		if (sscanf(line + offset, "%m[^ \t\n]s", &configuration[i]) != 1) {
 			configuration[i] = NULL;
 			break;
 		}
 
-		/* Ignore comments - everything between '#' and '\n' */
-		if (configuration[i][0] == '#') {
-			configuration[i] = NULL;
-			break;
+		len = strlen(configuration[i]);
+		if (configuration[i][0] == '"') {
```

Comment:
if an argument starts with the quotation mark, it means that it can contain spaces. configuration contains only one word without spaces. You need to find the next quotation mark and copy everything that is between these two marks. Have I missed something?

Reply:
No, you have not missed something. You are right. This problem exists in the parser since the beginning and should be fixed. This PR, however, was just trying to avoid crashes. I think this should be fixed, but not in this PR.

Reply:
Could you file an issue and add "// FIXME(issue/XXX): description" in the code

Reply:
I started to write the issue and then I just fixed it. Spaces should be now handled.

Reply:
I don't see how you fix it and my experiments show that it doesn't work...

Reply:
Do we describe the config format somewhere? Can we suppose that the first word is an option name and a rest of the line is an option argument?

Reply:
> I don't see how you fix it and my experiments show that it doesn't work...

It worked with my inputs. What do you use as input?

Reply:
```
$ cat test/others/config-file/conf.test
dsfa "aaaaa \"bbbbbb\""
```

I really recommend to implement the test for the parser. Otherwise it will never work reliably...

Reply:
implemented a unit test